M2Mqtt/Net/MqttNetworkChannel.cs

```diff
@@ -264,10 +281,38 @@ public void Connect()
                 if (this.clientCert != null)
                     clientCertificates = new X509CertificateCollection(new X509Certificate[] { this.clientCert });
 #if (NETSTANDARD1_6 || NETSTANDARD2_0 || NETCOREAPP3_1)
-                this.sslStream.AuthenticateAsClientAsync(this.remoteHostName,
-                    clientCertificates,
-                    MqttSslUtility.ToSslPlatformEnum(this.sslProtocol),
-                    false).Wait();
+
+                if ((this.alpnProtocols != null) && (0 < this.alpnProtocols.Count))
+                {
+                    this.sslStream = new SslStream(this.netStream, false);
+                    SslClientAuthenticationOptions authOptions = new SslClientAuthenticationOptions();
+                    List<SslApplicationProtocol> sslProtocolList = new List<SslApplicationProtocol>();
+                    foreach (string alpnProtocol in this.alpnProtocols)
+                    {
+                        sslProtocolList.Add(new SslApplicationProtocol(alpnProtocol));
+                    }
+                    authOptions.ApplicationProtocols = sslProtocolList;
+                    authOptions.EnabledSslProtocols = MqttSslUtility.ToSslPlatformEnum(this.sslProtocol);
+                    authOptions.TargetHost = remoteHostName;
+                    authOptions.AllowRenegotiation = false;
+                    authOptions.ClientCertificates = clientCertificates;
+                    authOptions.EncryptionPolicy = EncryptionPolicy.RequireEncryption;
+                    try
+                    {
+                        this.sslStream.AuthenticateAsClientAsync(authOptions).Wait();
+                    }
+                    catch(Exception ex)
+                    {
+                        throw ex;
+                    }
```

Comment:
What is the purpose of this try-catch if you want to re-throw the exception?

Reply:
It's useless try-catch. Sorry, I was checking AuthenticateAsClientAsync's Exception with "ex" on error case and missed to delete.
I delete this try-catch.